File: 690_employee_importance/employee_importance.py
from typing import List


class Employee:
    def __init__(self, id: int, importance: int, subordinates: List[int]):
        self.id = id
        self.importance = importance
        self.subordinates = subordinates


class Solution:
    def getImportance(self, employees: List['Employee'], id: int) -> int:
        importance, target = dict(), None
        for employee in employees:
            importance[employee.id] = employee
            if employee.id == id:
                target = employee
        queue = target.subordinates
        ans = target.importance
        while queue:
            _queue = []
            while queue:
                ep = importance[queue.pop()]
                ans += ep.importance
                _queue.extend(ep.subordinates)
            queue = _queue

        return ans

File: 690_employee_importance/employee_importance.java
/*
// Definition for Employee.
class Employee {
    public int id;
    public int importance;
    public List<Integer> subordinates;
};
*/

class Solution {
    public int getImportance(List<Employee> employees, int id) {
        HashMap<Integer, Employee> importance = new HashMap<Integer, Employee>();
        Employee target = null;
        for (Employee obj: employees) {
            importance.put(obj.id, obj);
            if (obj.id == id) {
                target = obj;
            }
        }
        int ans = target.importance;
        Queue<Integer> queue = new LinkedList<Integer>();
        for (int subId : target.subordinates) {
            queue.offer(subId);
        }
        while (!queue.isEmpty()) {
            Queue<Integer> queue_ = new LinkedList<Integer>();
            while (!queue.isEmpty()) {
                Employee obj = importance.get(queue.poll());
                ans += obj.importance;
                for (int subId : obj.subordinates) {
                    queue_.offer(subId);
                }
            }
            queue = queue_;
        }

        return ans;
    }
}